### PR TITLE
✨ Support daemonSet with AddOnTemplate

### PR DESCRIPTION
**Pull request by zhujian7**
## Related issue(s)

https://github.com/open-cluster-management-io/ocm/issues/368

Fixes #

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,8 @@ module open-cluster-management.io/ocm
 
 go 1.21
 
+replace open-cluster-management.io/addon-framework => github.com/zhujian7/addon-framework v0.0.0-20240614013034-40ee0596b536
```
Comment (zhujian7) [inline go.mod:5]: Remove after https://github.com/open-cluster-management-io/addon-framework/pull/275 merged

**Comment by haoqing0110** [comment]
LGTM

**Comment by zhujian7** [comment]
@zhiweiyin318 @haoqing0110 Some conflicts have just been resolved, PTAL.

**Pull request merged**